### PR TITLE
Issue #179 - Default base constructor calls should not be made - Analyzer, CodeFix

### LINKED ISSUE #179: Rule: Default base constructor calls should not be made
http://jira.sonarsource.com/browse/RSPEC-3253

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #179

### CODE REVIEW

File: src/Tests/SonarLint.UnitTest/TestCases/DefaultBaseConstructorCall.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        }
+        public MyClass(int p)
+        {
+
+        }
+    }
+
+    class DefaultBaseConstructorCall : MyClass
+    {
+        public DefaultBaseConstructorCall(): base()// Noncompliant
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/DefaultBaseConstructorCall.cs]: It seems slightly odd that comments at the end of line comments are removed.

File: src/Tests/SonarLint.UnitTest/TestCases/DefaultBaseConstructorCall.Fixed.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    class DefaultBaseConstructorCall : MyClass
+    {
+        public DefaultBaseConstructorCall()
+        {
+        }
+
+        public DefaultBaseConstructorCall(string s)
+
+        {
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/DefaultBaseConstructorCall.Fixed.cs]: It looks a bit weird with the whitespace left behind after the fix.

**Comment by duncanpMS** [comment]
LGTM

File: src/Tests/SonarLint.UnitTest/TestCases/DefaultBaseConstructorCall.cs (hunk 1 above)
Comment (tvsonar) [inline src/Tests/SonarLint.UnitTest/TestCases/DefaultBaseConstructorCall.cs]: @duncanpMS I reworked the whitespace removal based on your comments.

**Pull request merged**